### PR TITLE
chore: purge inverted index

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

during purging historical table data, purge inverted index data as well.

NOTE:

- dry run does NOT cover inverted index, will be addressed in dedicated PR (needs some refactoring)
- block with multiple inverted index files are not covered in this PR

  as inverted index is being refined / refactored, let's postpone this 

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15354)

### CODE REVIEW

File: src/query/storages/fuse/src/operations/gc.rs [hunk 1]
```diff
@@ -508,6 +544,13 @@ impl FuseTable {
                 .await?;
         }
 
+        let inverted_index_count = inverted_indexes_to_be_purged.len();
+        if inverted_index_count > 0 {
+            counter.inverted_indexes += inverted_index_count;
+            self.try_purge_location_files(ctx.clone(), inverted_indexes_to_be_purged)
```
Comment (sundy-li) [inline src/query/storages/fuse/src/operations/gc.rs:550]: There is also InvertedIndexFilterCache

Comment (dantengsky) [reply src/query/storages/fuse/src/operations/gc.rs:550]: thanks for reminding~~

File: src/query/storages/fuse/src/operations/gc.rs [hunk 2]
```diff
@@ -508,6 +588,16 @@ impl FuseTable {
                 .await?;
         }
 
+        let inverted_index_count = inverted_indexes_to_be_purged.len();
+        if inverted_index_count > 0 {
+            counter.inverted_indexes += inverted_index_count;
+            self.try_purge_location_files_and_cache::<InvertedIndexDirectory, _, _>(
```
Comment (b41sh) [inline src/query/storages/fuse/src/operations/gc.rs:594]: after PR #15362 merged, inverted index will be readed as `InvertedIndexMeta` and multiple `InvertedIndexFile`s, including:
1. fast file
2. store file
3. fieldnorm file
4. position file
5. idx file
6. term file
7. meta.json file
8. .managed.json file

The cache key is a combination of a file path and a subfile name,
for example, if the inverted index file path is `1/23819/_i_i/idx1/307a34e/cf1ecaf64ff541bb8a2919df86e73d80_v0.index`
the cache key of each subfile is 
`23819/_i_i/idx1/307a34e/cf1ecaf64ff541bb8a2919df86e73d80_v0.index-fast`
`23819/_i_i/idx1/307a34e/cf1ecaf64ff541bb8a2919df86e73d80_v0.index-store`
...

Comment (dantengsky) [reply src/query/storages/fuse/src/operations/gc.rs:594]: got it, thanks

Comment (dantengsky) [reply src/query/storages/fuse/src/operations/gc.rs]: @b41sh  please take another look

**Review by b41sh — APPROVED** [review]

**Pull request merged**